fsck/fsck.c

```diff
@@ -215,6 +226,39 @@ static int check_clus_chain(struct exfat_de_iter *de_iter,
 	}
 
 	return 0;
+locate_cluster:
+	node->size = count * exfat->clus_size;
+	exfat_de_iter_get_dirty(de_iter, 1, &stream_de);
+	err = find_free_cluster(exfat, exfat->start_clu, &new_clus);
+
+	if (err) {
+		exfat->start_clu = EXFAT_FIRST_CLUSTER;
+		exfat_err("failed to find a free cluster\n");
+		return -ENOSPC;
+	}
+	exfat->start_clu = new_clus;
+
+	if (exfat_set_fat(exfat, new_clus, EXFAT_EOF_CLUSTER))
+		return -EIO;
+
+	/* zero out the new cluster */
+	if (exfat_write(exfat->blk_dev->dev_fd, exfat->zero_cluster,
+			exfat->clus_size, exfat_c2o(exfat, new_clus)) !=
+	    (ssize_t)exfat->clus_size) {
+		exfat_err("failed to fill new cluster with zeroes\n");
+		return -EIO;
+	}
+	exfat_info("final cluster : %#x \n", new_clus);
+	count = 1;
+	stream_de->stream_start_clu = cpu_to_le64(new_clus);
+	stream_de->stream_size = cpu_to_le64(count * exfat->clus_size);
+	stream_de->stream_valid_size = cpu_to_le64(count * exfat->clus_size);
+	stream_de->dentry.stream.flags = 3;
```

Comment:
Could you use macro constant instead of magic value?

Reply:
OK.